### PR TITLE
Remove CS-active VMThread flag

### PR DESCRIPTION
**Pull request by amicic**
All users of CS-active flag rely on read barrier range fields in
VMThread.

Signed-off-by: Aleksandar Micic <amicic@ca.ibm.com>

### CODE REVIEW

**Comment by amicic** [comment]
@fjeremic I found all but one site in JIT has been removed, which is just an assert. I tool liberty of removing it myself.

**Comment by AdamBrousseau** [comment]
Jenkins copyright check

File: runtime/compiler/env/VMJ9.cpp [hunk 1]
```diff
@@ -1908,8 +1908,6 @@ UDATA TR_J9VMBase::thisThreadGetGSIntermediateResultOffset()
  */
 UDATA TR_J9VMBase::thisThreadGetConcurrentScavengeActiveByteAddressOffset()
    {
-   TR_ASSERT_FATAL(J9_PRIVATE_FLAGS_CONCURRENT_SCAVENGER_ACTIVE == 0x20000,
-              "GSCS: The check weither concurrent scavenge is active is dependant on the flag being 0x20000");
```
Comment (fjeremic) [inline runtime/compiler/env/VMJ9.cpp]: @NigelYiboYu can you confirm this function is no longer used?

**Comment by fjeremic** [comment]
> @fjeremic I found all but one site in JIT have been removed, which is just an assert. I took the liberty of removing it myself.

Thanks, yes and let's leave #5458 open to track the removal of the unused APIs.

**Pull request merged**